src/base/bittorrent/peerinfo.cpp

```diff
@@ -408,3 +408,8 @@ QString PeerInfo::flagsDescription() const
 {
     return m_flagsDescription;
 }
+
+int PeerInfo::downloadingPieceIndex() const
```

Comment:
By the way, I hope you will not use `TorrentInfo::filesForPiece()` in the GUI code directly? You can implement similar method `TorrentHandle::downloadingFiles()`. Although it is difficult for me to judge what I have not seen. We'll see.